__init__.py

```diff
@@ -286,20 +298,30 @@ def load_nickname():
     CHANGE_YEAR: ctrl_chg_yrs,
     CHANGE_MONTH: ctrl_chg_mth,
     CHANGE_DAY: ctrl_chg_day,
+    CHANGE_THEME: ctrl_chg_thm,
+    CHANGE_FGCOLOR: ctrl_chg_fgc,
+    CHANGE_BGCOLOR: ctrl_chg_bgc,
 }
 
 def main():
+    global conf
+    global theme
     global updated
     try:
         load_nickname()
+        conf = config.Config()
+        theme = THEMES[conf.themeid]
         with display.open() as d:
             last_secs, secs = 0, 0
             last_msecs, msecs = 0, 0
             while True:
                 updated = False
 
                 bs = checkButtons()
+                saveConfig = (MODE != DISPLAY)
                 CTRL_FNS[MODE](bs)
+                if saveConfig and MODE == DISPLAY:
+                    conf.writeConfig() # store config on leaving settings
```

Comment:
It's probably cleaner to do this in the `ctrl` functions that change configuration, but as the code is laid out currently, that would be quite a bit of repetition, so without rewriting the mode system, this is probably the cleanest option.

Once the mode system _does_ get rewritten, this should probably move into the mode functions.
(I figured having classes for modes and having the ones that inherit from `ChangesConfig` save their config on exit or something like that)

Reply:
First I had the configuration writing in the `chg_` functions itself (after all it is only one repetitive line of code), but I wanted to preserve flash write cycles and so I thought it makes sense to only write on exit of the settings.